core/thread/src/TWin32Mutex.cxx

```diff
@@ -69,3 +69,55 @@ Int_t TWin32Mutex::UnLock(void)
    ::LeaveCriticalSection(&fCritSect);
    return 0;
 }
+
+namespace {
+struct TWin32MutexState : public TVirtualMutex::State {
+   int fLockCount = 0;
+};
+}
+
+////////////////////////////////////////////////////////////////////////////////
+/// Reset the mutex state to unlocked. The state before resetting to unlocked is
+/// returned and can be passed to `Restore()` later on. This function must only
+/// be called while the mutex is locked.
+
+std::unique_ptr<TVirtualMutex::State> TWin32Mutex::Reset()
+{
+   std::unique_ptr<TWin32MutexState> pState(new TWin32MutexState);
+   if (TestBit(kIsRecursive)) {
+      while (!UnLock())
+         ++pState->fLockCount;
+      return std::move(pState);
```

Comment:
Using std::move() here is probably not what you want: https://stackoverflow.com/a/11817902/1034717.

Reply:
I don't know, I took the code from existing `TPosixMutex::Reset()`...

Reply:
Ah, I see. Two places we will have to change, then... The std::move() on the return value prevents some compiler optimizations.

Reply:
Indeed, my fault - thanks for spotting that!

Reply:
Ok, I will merge and then remove the `std::move()` later from both places.